cpp/log/cert.cc

```diff
@@ -278,21 +307,22 @@ Cert::Status Cert::HasBasicConstraintCATrue() const {
     LOG(ERROR) << "Failed to check BasicConstraints extension";
   }
 
-  if (status != TRUE)
-    return status;
+  if (status != TRUE) {
+    return CertStatusToStatusOrBool(status);
```

Comment:
Yes, the Cert::Status isn't helping either. There are various OpenSSL errors on the way to getting the extension. Might be worth another pass after this is done as we probably don't care exactly how it went wrong in OpenSSL.